src/core/modules/engines/engines_gamerules.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+* =============================================================================
+* Source Python
+* Copyright (C) 2012-2020 Source Python Development Team.  All rights reserved.
+* =============================================================================
+*
+* This program is free software; you can redistribute it and/or modify it under
+* the terms of the GNU General Public License, version 3.0, as published by the
+* Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+* details.
+*
+* You should have received a copy of the GNU General Public License along with
+* this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* As a special exception, the Source Python Team gives you permission
+* to link the code of this program (as well as its derivative works) to
+* "Half-Life 2," the "Source Engine," and any Game MODs that run on software
+* by the Valve Corporation.  You must obey the GNU General Public License in
+* all respects for all other code used.  Additionally, the Source.Python
+* Development Team grants this exception to all derivative works.
+*/
+
+//-----------------------------------------------------------------------------
+// Includes.
+//-----------------------------------------------------------------------------
+// Source.Python
+#include "engines_gamerules.h"
+#include "modules/entities/entities_props.h"
+#include "modules/stringtables/stringtables.h"
+#include "utilities/wrap_macros.h"
+
+// Boost.Python
+#include "boost/python.hpp"
+using namespace boost::python;
+
+// SDK
+#include "server_class.h"
+
+
+//---------------------------------------------------------------------------------
+// External variables to use.
+//---------------------------------------------------------------------------------
+extern INetworkStringTableContainer *networkstringtable;
+
+
+//-----------------------------------------------------------------------------
+// Functions
+//-----------------------------------------------------------------------------
+int find_game_rules_property_offset(const char* name)
+{
+	ServerClass* cls = ServerClassExt::find_server_class(find_game_rules_proxy_name());
```

Comment:
Since the proxy name is cached into `find_game_rules_proxy_name`, then we could also cache `cls` here (or even better, the table itself) so that it doesn't have to be resolved every times an offset is looked up.

Reply:
Yeah, I wasn't sure if we can cache that. I will add that later today.

Reply:
Done! :)